includes/admin/auth-service-actions.php

```diff
@@ -60,69 +62,88 @@ public function oauth_deauthenticate_site()
 			'method' => 'POST',
 			'body' => $send_data,
-			'cookies' => array()
-		   ));
-
-    if ( is_wp_error( $request ) || wp_remote_retrieve_response_code( $request ) != 200 ) {
-        error_log( print_r( $request, true ) );
-    }
-
-    $response = wp_remote_retrieve_body( $request );
-
-		$error_msg = array();
-	if($response  == true || $response >= 1){
-		delete_option('simple_calendar_auth_site_token');
-		$message = __('DeAuthenticate Successfully.', 'google-calendar-events');
-		$send_msg = array('message' => $message);
-		wp_send_json_success($send_msg);
-	   }else{
-		if(isset($message['message']) && !empty($message['message'])){
-			$message = $message['message'];
-		}else{
-			$message = __('DeAuthentication Faild.', 'google-calendar-events');
+			'cookies' => [],
+		]);
+
+		if (is_wp_error($request) || wp_remote_retrieve_response_code($request) != 200) {
+			error_log(print_r($request, true));
 		}
 
-		$error_msg = array('message' => $message);
-		wp_send_json_error($error_msg);
-	   }
-		die();
+		$response = wp_remote_retrieve_body($request);
 
-    }
+		$error_msg = [];
+		if ($response == true || $response >= 1) {
+			delete_option('simple_calendar_auth_site_token');
+			$message = __('DeAuthenticate Successfully.', 'google-calendar-events');
+			$send_msg = ['message' => $message];
+			wp_send_json_success($send_msg);
+		} else {
+			if (isset($message['message']) && !empty($message['message'])) {
+				$message = $message['message'];
+			} else {
+				$message = __('DeAuthentication Faild.', 'google-calendar-events');
+			}
+
+			$error_msg = ['message' => $message];
+			wp_send_json_error($error_msg);
+		}
+		die();
+	}
 
 	/*
-	* Check if token expire
-	*/
-	public function oauth_check_iftoken_expired(){
-		$send_data = array(
+	 * Check if token expire
+	 */
+	public function oauth_check_iftoken_expired()
+	{
+		$send_data = [
 			'site_url' => self::$my_site_url,
 			'auth_token' => get_option('simple_calendar_auth_site_token'),
-		);
-		$request = wp_remote_post(self::$url.'check_iftoken_expired', array(
+		];
+		$request = wp_remote_post(self::$url . 'check_iftoken_expired', [
 			'method' => 'POST',
 			'body' => $send_data,
-			'cookies' => array()
-		));
+			'cookies' => [],
+		]);
 
-		$response = wp_remote_retrieve_body( $request );
-
-		if($response  == true || $response >= 1){
+		$response = wp_remote_retrieve_body($request);
 
+		if ($response == true || $response >= 1) {
 			return 'valid';
-
-		}else{
+		} else {
 			delete_option('simple_calendar_auth_site_token');
 			return 'invalid';
 		}
+	}
+
+	/*
+	 * Get calendar list
+	 */
+	public function auth_get_calendarlist()
+	{
+		$send_data = [
+			'site_url' => self::$my_site_url,
+			'auth_code' => '',
+		];
+		$request = wp_remote_post(self::$url . 'auth_get_calendarlist', [
+			'method' => 'POST',
+			'body' => $send_data,
+			'cookies' => [],
+		]);
+
+		$response = wp_remote_retrieve_body($request);
```

Comment:
What will happen if response is unsuccessful?

Reply:
Code updated